packages/snap-tracker/src/Tracker.ts

```diff
@@ -174,6 +174,110 @@ export class Tracker extends Beacon {
 				});
 			}
 		});
+
+		const cart = (this.globals as TrackerGlobals).cart;
+		if (Array.isArray(cart)) {
+			if (cart.length === 0) {
+				// cart is empty, clear storage and send remove event if storage had items
+				const storedCart = this.storage.cart.get();
+				if (storedCart.length) {
+					this.events.cart.remove({
+						data: {
+							results: storedCart,
+							cart: [],
+						},
+					});
+				}
+				this.storage.cart.clear();
+			} else if (cart.length) {
+				// length check here to be able to sent error event if invalid array of objects is provided
+				const currentCart: Product[] = cart
+					.filter((item) => typeof item === 'object' && (item.uid || item.sku || item.childUid || item.childSku) && item.qty && item.price)
+					.map((item): Product => {
```

Comment:
I'm not sure I understand the map usage here - seems to map to same properties as `item`... 🤔